lib/transmitter.js

```diff
@@ -31,14 +32,22 @@ function transmitEvents(url, projectId, agentId) {
     .catch((error) => {
       debug(`Error transmitting events: ${error}`);
     });
-  eventsToSend = [];
+  clearEvents();
   return postPromise;
 }
 
+function clearEvents() {
+  eventsToSend.length = 0;
```

Comment:
this seems to be best way to clear arrays (without redefining them), and it looks like it's clearing the memory:
```> const myArray = [1,2,3,4,5];
undefined
> myArray
[ 1, 2, 3, 4, 5 ]
> myArray.length = 0;
0
> myArray
[]
> myArray[0]
undefined```
a bit hard to precisely measure the memory consumption of the process since it seems unstable regardless of my variable definitions.